monosi/anomalies/__init__.py

```diff
@@ -27,19 +36,30 @@ def _std_dev(cls, values: List[Any]):
         return std_dev
 
     @classmethod
-    def _z_scores(cls, stats: List[Any]):
-        values = [stat.value for stat in stats]
+    def _z_scores(cls, stats: List[MetricStat], filter_null=True):
+        stats_to_calc = stats
 
-        mean = cls._mean(values)
-        std_dev = cls._std_dev(values)
+        if filter_null:
+            stats_to_calc = cls._remove_nulls(stats)
+
+        values = [stat.value for stat in stats_to_calc]
+
+        try:
+            mean = cls._mean(values)
+            std_dev = cls._std_dev(values)
+        except ZeroDivisionError:
+            return []
 
         if (std_dev == 0): return []
 
-        for stat in stats:
+        for stat in stats_to_calc:
             stat.std_dev = std_dev
-            stat.z_score = round(((stat.value - mean) / std_dev), 2) 
+            try:
+                stat.z_score = round(((stat.value - mean) / std_dev), 2) 
+            except TypeError:
+                return []
```

Comment:
Why do we need a try catch here?

Reply:
because `stat.value` has the possibility of being None and if so we don't want it to break on calculation

Reply:
this special casing is going to be pain everywhere but I buy that.